brainscore_core/submission/endpoints.py

```diff
@@ -0,0 +1,210 @@
+"""
+Process plugin submissions (data, metrics, benchmarks, models) and score models on benchmarks.
+"""
+import traceback
+import logging
+from abc import ABC
+from datetime import datetime
+import http
+import os
+import random
+import requests
+from requests.auth import HTTPBasicAuth
+import string
+from typing import List, Union, Dict
+
+from brainscore_core import Benchmark, Score
+from brainscore_core.submission import database_models
+from brainscore_core.submission.database import connect_db, modelentry_from_model, \
+    submissionentry_from_meta, benchmarkinstance_from_benchmark, update_score, \
+    public_model_identifiers, public_benchmark_identifiers, uid_from_email, email_from_uid
+
+logger = logging.getLogger(__name__)
+
+
+def process_github_submission(plugin_info: Dict[str, Union[List[str], str]]):
+    """
+    Triggered when changes are merged to the GitHub repository, if those changes affect benchmarks or models.
+    Starts parallel runs to score models on benchmarks (`run_scoring`).
+    """
+    jenkins_base = "http://braintree.mit.edu:8080"
+    jenkins_usr = os.environ['JENKINS_USR']
+    jenkins_token = os.environ['JENKINS_TOKEN']
+    jenkins_trigger = os.environ['JENKINS_TRIGGER']
+    jenkins_job = "dev_score_plugins"
+
+    url = f'{jenkins_base}/job/{jenkins_job}/buildWithParameters?token={jenkins_trigger}'
+    payload = {k:v for k,v in plugin_info.items() if plugin_info[k]}
+    auth_basic=HTTPBasicAuth(username=jenkins_usr, password=jenkins_token)
+    r = requests.get(url, params=payload, auth=auth_basic)
+    print(r)
+
+
+def get_email_from_uid(uid: int) -> str:
```

Comment:
what's this method used by? Right now it looks like a redundant addition

Reply:
It's invoked from GitHub Actions to get a user's email if their web-submitted PR fails, so I figured it's polite to wrap the database call in an endpoint method.